File: concesiones/concesiones/report/ventas_de_abastecimiento_interno/ventas_de_abastecimiento_interno.js
// Copyright (c) 2016, Marco Bohorquez and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Ventas de Abastecimiento Interno"] = {
	"filters": [
		{
			"fieldname": "customer",
			"label": __("Cliente"),
			"fieldtype": "Link",
			"options": "Customer",
			"default": "",
			"width": "150px",
		},
		{
			"fieldname": "sucursal",
			"label": __("Sucursal"),
			"fieldtype": "Link",
			"options": "Branch",
			"width": "100px",
			"default": "",
			"get_query": function() {
				return {
					"filters": {
						"concesionario": 1
					}
				};
			}
		},
		{
			"fieldname": "month",
			"label": __("Mes"),
			"fieldtype": "Select",
			"default": "5",
			"options": [
				{ "value": "todos", "label": __("") },
				{ "value": "1", "label": __("Enero") },
				{ "value": "2", "label": __("Febrero") },
				{ "value": "3", "label": __("Marzo") },
				{ "value": "4", "label": __("Abril") },
				{ "value": "5", "label": __("Mayo") },
				{ "value": "6", "label": __("Junio") },
				{ "value": "7", "label": __("Julio") },
				{ "value": "8", "label": __("Agosto") },
				{ "value": "9", "label": __("Septiembre") },
				{ "value": "10", "label": __("Octubre") },
				{ "value": "11", "label": __("Noviembre") },
				{ "value": "12", "label": __("Diciembre") }
			],
			"width": "100px"
		},
		{
			"fieldname": "year",
			"label": __("Año"),
			"fieldtype": "Select",
			"options": [
				{ "value": "2023", "label": __("2023") },
				{ "value": "2024", "label": __("2024") },
				{ "value": "2025", "label": __("2025") }
			],
			"width": "100px",
			"default": "2024"
		},
	]
};

File: concesiones/concesiones/report/ventas_de_abastecimiento_interno/ventas_de_abastecimiento_interno.py
# Copyright (c) 2013, Frappe Technologies Pvt. Ltd. and contributors
# For license information, please see license.txt

# import frappe
import requests
import frappe
from datetime import timedelta
import calendar


def execute(filters=None):
	columns, data = [], []
	columns = get_columns()

	if filters.get("customer") is not None:
		customer = filters.get("customer")
	else:
		customer = "todos"

	if filters.get("sucursal") is not None:
		sucursal = filters.get("sucursal")
	else:
		sucursal = "todos"

	if filters.get("month") is not None:
		month = filters.get("month")
	else:
		month = "todos"

	if filters.get("year") is not None:
		year = filters.get("year")
	else:
		year = "todos"

	url = "https://horario-salida-qa-erpwin.shalom.com.pe/ventas-abastecimiento-interno/" + customer + "/" + sucursal + "/" + month + "/" + year

	arrayData = requests.get(url)
	arrayData = arrayData.json()

	if "message" in arrayData:
		return columns, arrayData.get("message")

def get_columns():
	columns = [
		{
			'label': "Cliente",
			'fieldname': 'customer',
			'fieldtype': 'Data',
			'width': 100
		},
		{
			'label': "Agencia",
			'fieldname': 'agencia',
			'fieldtype': 'Data',
			'width': 150,
		},
		{
			'label': "Fecha",
			'fieldname': 'posting_date',
			'fieldtype': 'Data',
			'width': 120,
		},
		{
			'label': "Total",
			'fieldname': 'total',
			'fieldtype': 'Data',
			'width': 60,
		},
		{
			'label': "Serie",
			'fieldname': 'series',
			'fieldtype': 'Data',
			'width': 60,
		},
		{
			'label': "Numero",
			'fieldname': 'numeros',
			'fieldtype': 'Data',
			'width': 80,
		},
		{
			'label': "Link Comprobante",
			'fieldname': 'link_pdf',
			'fieldtype': 'Data',
			'width': 150,
		},
		{
			'label': "Id Nota Entrega",
			'fieldname': 'id_nota_entrega',
			'fieldtype': 'Data',
			'width': 150,
		},
		{
			'label': "Id Factura Venta",
			'fieldname': 'id_factura_venta',
			'fieldtype': 'Data',
			'width': 150,
		}
	]
	return columns
